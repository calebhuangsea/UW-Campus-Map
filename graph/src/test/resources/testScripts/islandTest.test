# Test having self edge and cycle edge in one graph connected
# Test that a graph can be created
# 4 nodes can be added
# a cycle graph can be formed
# a self edge can be formed

# Create a graph
CreateGraph graph1

# Add four nodes
AddNode graph1 n4
AddNode graph1 n3
AddNode graph1 n2
AddNode graph1 n1

# Add cycle edge
AddEdge graph1 n1 n1 e1
AddEdge graph1 n2 n3 e2
AddEdge graph1 n3 n4 e3
AddEdge graph1 n4 n2 e4
AddEdge graph1 n1 n2 e5

# Print the node in the graph and the outgoing edges from n1 n2 n3 n4
ListNodes graph1
ListChildren graph1 n1
ListChildren graph1 n2
ListChildren graph1 n3
ListChildren graph1 n4
