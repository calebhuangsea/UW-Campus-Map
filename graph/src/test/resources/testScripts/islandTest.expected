# Test having self edge and cycle edge in one graph connected
# Test that a graph can be created
# 4 nodes can be added
# a cycle graph can be formed
# a self edge can be formed

# Create a graph
created graph graph1

# Add four nodes
added node n4 to graph1
added node n3 to graph1
added node n2 to graph1
added node n1 to graph1

# Add cycle edge
added edge e1 from n1 to n1 in graph1
added edge e2 from n2 to n3 in graph1
added edge e3 from n3 to n4 in graph1
added edge e4 from n4 to n2 in graph1
added edge e5 from n1 to n2 in graph1

# Print the node in the graph and the outgoing edges from n1 n2 n3 n4
graph1 contains: n1 n2 n3 n4
the children of n1 in graph1 are: n1(e1) n2(e5)
the children of n2 in graph1 are: n3(e2)
the children of n3 in graph1 are: n4(e3)
the children of n4 in graph1 are: n2(e4)